ethereum/core/src/test/java/tech/pegasys/pantheon/ethereum/core/TransactionPoolTest.java

```diff
@@ -355,6 +359,35 @@ public void shouldNotNotifyBatchListenerIfNoTransactionsAreAdded() {
     verifyZeroInteractions(batchAddedListener);
   }
 
+  @Test
+  public void shouldAllowWhitelistedTransactionWhenWhitelistEnabled() {
+    transactionPool.setAccountWhitelist(accountWhitelistController);
+    givenTransactionIsValid(transaction1);
+
+    when(accountWhitelistController.isAccountWhiteListSet()).thenReturn(true);
+    when(accountWhitelistController.contains(transaction1.getSender().toString())).thenReturn(true);
+
+    assertThat(transactionPool.addLocalTransaction(transaction1)).isEqualTo(valid());
+
+    assertTransactionPending(transaction1);
+  }
+
+  @Test
+  public void shouldRejectNonWhitelistedTransactionWhenWhitelistEnabled() {
+    transactionPool.setAccountWhitelist(accountWhitelistController);
+    givenTransactionIsValid(transaction1);
+
+    when(accountWhitelistController.isAccountWhiteListSet()).thenReturn(true);
+    when(accountWhitelistController.contains(transaction1.getSender().toString()))
+        .thenReturn(false);
+
+    assertThat(transactionPool.addLocalTransaction(transaction1))
+        .isEqualTo(ValidationResult.invalid(TX_SENDER_NOT_AUTHORIZED));
+
+    assertTransactionNotPending(transaction1);
+    verifyZeroInteractions(batchAddedListener);
+  }
+
```

Comment:
We should probably have a unit test to verify the behaviour when the `accountWhitelistController` is not present. Something like `shouldAllowTransactionWhenAccountWhitelistControllerIsNotPresent()`.

Reply:
actioned